src/components/PrivateRoute.js

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { Route, Redirect } from 'react-router-dom';
+
+const PrivateRoute = ({ component: Component, ...rest }) => {
+  return (
+    <Route
+      {...rest}
+      render={props => {
+        if (localStorage.getItem('token')) {
+          return <Component {...props} />;
+        }
+        alert('Hi there 😊, Please Sign in');
```

Comment:
@Omulosi Can we remove the alert, instead have a message put in the redirect state so if there is a message the login page will render it